test/mithril.js

```diff
@@ -37,18 +37,50 @@ describe("m()", function () {
 		expect(m("[title=bar]")).to.have.deep.property("attrs.title", "bar")
 	})
 
+	it("sets correct unquoted attr with unicode characters", function () {
+		expect(m("[title=bö💥]")).to.have.deep.property("attrs.title", "bö💥")
+	})
+
 	it("sets attr without a value as an empty string", function () {
 		expect(m("[empty]")).to.have.deep.property("attrs.empty", "")
 	})
 
 	it("sets correct single quoted attr", function () {
-		expect(m("[title=\'bar\']")).to.have.deep.property("attrs.title", "bar")
+		expect(m("[title='bar']")).to.have.deep.property("attrs.title", "bar")
+	})
+
+	it("sets correct single quoted attr with old-style embedded single quote", function () {
+		expect(m("[title='b'ar']")).to.have.deep.property("attrs.title", "b'ar")
+	})
+
+	it("sets correct single quoted attr with an embedded single quote", function () {
+		expect(m("[title='b\\'ar']")).to.have.deep.property("attrs.title", "b'ar")
+	})
+
+	it("sets correct single quoted attr with an embedded escape sequences and quotes", function () {
+		expect(m("[title='b\\\\\\'\\\"\\\\\"ar']")).to.have.deep.property("attrs.title", "b\\'\"\\\"ar")
+	})
+
+	it("sets correct single quoted attr with an embedded closing bracket", function () {
+		expect(m("[title='b]ar']")).to.have.deep.property("attrs.title", "b]ar")
 	})
 
 	it("sets correct double quoted attr", function () {
 		expect(m("[title=\"bar\"]")).to.have.deep.property("attrs.title", "bar")
 	})
 
+	it("sets correct double quoted attr with old-style embedded double quotes", function () {
+		expect(m("[title=\"b\"a\"r\"]")).to.have.deep.property("attrs.title", "b\"a\"r")
```

Comment:
This should also fail to parse. The selector is seen as `[title="b"a"r"]`, which is syntactically invalid.